src/core/dom/index.js

```diff
@@ -557,3 +557,17 @@ export function getChildJsonConfig(element) {
 export function isServerRendered(element) {
   return element.hasAttribute('i-amphtml-ssr');
 }
+
+/**
+ * Propagate the nonce found in <head> to a new script element.
+ * Recent browsers force nonce to be accessed via property instead of attribute.
+ *
+ * @param {Document} doc
+ * @param {HTMLScriptElement} scriptEl
+ */
+export function propagateNonce(doc, scriptEl) {
+  const currentScript = doc.head.querySelector('script[nonce]');
+  if (currentScript) {
+    scriptEl.nonce = currentScript.nonce ?? currentScript.getAttribute('nonce');
```

Comment:
Should this be using `setAttribute` so we continue to support older browsers?

Reply:
Is setting the property unsupported in old browsers?

Reply:
I'll make it setAttribute just in case, since that is known to work